src/appleseed/renderer/modeling/bsdf/blinnbrdf.cpp

```diff
@@ -291,4 +294,28 @@ auto_release_ptr<BSDF> BlinnBRDFFactory::create(
     return auto_release_ptr<BSDF>(new BlinnBRDF(name, params));
 }
 
+//
+// MicrofacetBlinnBRDFFactory class implementation.
+//
+
+const char* MicrofacetBlinnBRDFFactory::get_model() const
+{
+    return MicrofacetModel;
+}
+
+Dictionary MicrofacetBlinnBRDFFactory::get_model_metadata() const
+{
+    return
+        Dictionary()
+            .insert("name", MicrofacetModel)
+            .insert("label", "Microfacet Blinn BRDF");
+}
+
```

Comment:
Is there no `MicrofacetBlinnBRDFFactory::get_input_metadata()` needed, as it is for the regular BRDF? This comment applies to all other BRDFs too.